ONSdigital/ras-frontstage
# pip-tools test

# Motivation and Context
Testing pip-tools as an alternative to pipenv, for "Spike alternatives to pipenv".


# What has changed
- Converted Pipfile to requirements.in and dev-requirements.in using `pipenv2tools`
- Generated requirements.txt and dev-requirements.txt to replace Pipfile and Pipfile.lock
- Swapped pipenv command for `pip -r`
- added pip package safety to replace pipenv check: https://pypi.org/project/safety/
- removed old Pipfile and Pipfile.lock
- Removed obsolete flask-zipkin last, to show changes to requirements files in the diffs





# How to test?
**Local**: `make unit-tests`; `make integration-tests`. 
If you want to test adding or removing a package from requirements.in (or dev-requirements.in), `pip install -U pip-tools`, then `pip-compile --generate-hashes dev-requirements.in`. Note that the main req file gets compiled first, followed by the dev req file which invokes the main one.

**Dev**: spin up a frontstage pod with the "using-piptools" image, and run acceptance tests against it.




# Links
https://trello.com/c/VgmfNC6T




# Screenshots (if appropriate):
